test/cli_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"github.com/alibaba/pouch/test/command"
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/go-check/check"
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+// PouchSuite is the test suite for pouch cli.
+type PouchSuite struct{}
+
+func init() {
+	check.Suite(&PouchSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *PouchSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+}
+
+// TestImageTagOKWithSourceImageName tests OK.
+func (suite *PouchSuite) TestInterspersedFlags(c *check.C) {
+	// set flags directly to pouch cli
+	command.PouchRun("-D", "ps").Assert(c, icmd.Success)
```

Comment:
```
cli_test.go:25:
    // set flags directly to pouch cli
    command.PouchRun("-D", "ps").Assert(c, icmd.Success)
/home/travis/gopath/src/github.com/alibaba/pouch/vendor/github.com/gotestyourself/gotestyourself/icmd/command.go:61:
    t.Fatalf("at %s:%d - %s\n", filepath.Base(file), line, err.Error())
... Error: at cli_test.go:25 - 
Command:  /usr/local/bin/pouch -D ps
ExitCode: 1
Error:    exit status 1
Stdout:   
Stderr:   Error: unknown shorthand flag: 'D' in -D
```

hmm, weird behaviour...

Reply:
I agree. I think it is cobra does not support this. While moby project hacked on this.